### DISCUSSION TITLE
Hubs clouds on .space domains have functionality blocked by uBlock firefox add-on

### DISCUSSION THREAD

**Discussion by wmurphyrd** [mozilla/hubs#3606] [Hubs Cloud Questions] [answered]
I have an AWS hubs cloud running a custom client. I had previously enabled cloudflare CDN successfully, but I wanted to disable it due to an incompatibility I discovered with my custom client. I logged in to the admin panel, unchecked the use cloudflare worker box and saved. 

Immediately, all pages on my hubs cloud ceased to load. I saw only a blank page and the console included several CSP errors suggesting it was still trying to load content from the workers. Here's an example of one.

```
Content Security Policy: The page’s settings blocked the loading of a resource at https://hubs-vreign-internal-space-proxy.immers-space.workers.dev/hubs/assets/js/frontend-12728b92ac4a86430db1.js (“script-src”).
```

At this point I tried the following troubleshooting steps, but none of them restored the hub:

1. Updating stack
2. Terminating ec2 instance to restart
3. [Searching the parameter store for values related to workers](https://github.com/mozilla/hubs/discussions/3138) (there were none)

Here, I would have been stuck, but luckily a teammate had login credentials in his local console (I could not login because the login page also failed to load upon clicking the email link). He redeployed the custom client and the hub became accessible again, but not fully functional. It will load on desktop, but the physics worker fails to load and reports a CORS error in the console. 

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://hub-vreign-assets.vreign-internal.space/hubs/assets/wasm/ammo.wasm-8640fe8d5bf91d2e84ea56e0c1307ba9.wasm. (Reason: CORS request did not succeed).
```

The consequence of this is physics doesn't function on desktop (spawned objects do not move when grabbed), and the room becomes totally inaccessible in SteamVR (locks up after showing first frame and falls back to SteamVR area like a very low framerate). However, it works properly in FFxR on Quest2 (perhaps with reduced performance, not certain on this)

Here are the response headers I get if I curl that ammo.wasm url (the file is served successfully)

```
HTTP/2 200 
content-type: application/wasm
content-length: 845592
date: Sat, 19 Dec 2020 16:25:03 GMT
last-modified: Sat, 19 Dec 2020 16:15:34 GMT
etag: "8640fe8d5bf91d2e84ea56e0c1307ba9"
cache-control: public, max-age=31536000
accept-ranges: bytes
server: AmazonS3
x-cache: Miss from cloudfront
via: 1.1 ba9c4a15d904b10c426461fa8ff12179.cloudfront.net (CloudFront)
x-amz-cf-pop: ORD53-C1
```

Here are additional troubleshooting steps I've tried from this point, I have not been able to restore full functionality

1. Undeploy custom client
2. Update stack
2. Remove my advanced settings CORS setting (`*`) (and after this did not fix, readded)
3. Unregistered `hub.service.js` the service worker to force update
4. Terminating/restarting EC2
5. Undeploying cloudflare workers (I could see in the cloudflare dashboard that my workers for this domain were still getting requests)

After undeploying the cloudflare workers, I started getting new console errors - it is still trying to use the CDN workers:

```
Error adding media Error: Error resolving url "https://hubs-vreign-internal-space-proxy.immers-space.workers.dev/files/c7f00b96-a148-4628-be45-419e081ceff5.jpg": Internal Server Error
    t media-utils.js:54
...
   media-loader.js:627:6
```

I don't have any other ideas before nuking the entire stack and starting over

**Comment by robinkwilson** [comment, marked as answer]
Thank you VERY much for the detailed steps above. I'm happy to help you get unstuck.

So glad you were able to redeploy the custom client, definitely necessary.

For the `Error adding media .jpg`, **you'll need to add that image again.** This time it will not save with the `https://hubs-vreign-internal-space-proxy.immers-space.workers.dev` proxy since that is disabled.

For the `Cross-Origin Request Blocked: The Same Origin Policy`, you'll need to go into the admin panel (my answer for this discussion thread shows what I added to make the CORS origin policy to work: https://github.com/mozilla/hubs/discussions/2890)

Append this to the CORS settings **in addition** to your main domain, internal domain, link domain (as outlined in the linked discussion):`https://hub-vreign-assets.vreign-internal.space`

**Reply by wmurphyrd** [reply]
Thanks @robinkwilson. While these  CORS parameter solutions did not work for me, I found the true issue and it wasn't Hubs's fault at all. So sorry. In a case of monumentally bad timing, it seems the uBlock Origin firefox plugin decided to start blocking all `wasm` scripts from `.space` domains right when I was disabling my CDN :((((

![image](https://user-images.githubusercontent.com/10034859/102811658-50338180-438b-11eb-91cc-e04ad06eda50.png)

For me, disabling uBlock origin for my hubs site restored functionality. Not sure what to do about this for my visitors, though

Also thanks for the tip on needing to republish scenes to get the CDN links out of there. Will try that not that I can get publishing working again (another `wasm` was being blocked there)

**Reply by wmurphyrd** [reply]
Update! The lovely folks at uBlock have remove this filter https://github.com/uBlockOrigin/uAssets/issues/8389
